app.js

```diff
@@ -1,55 +1,163 @@
+    console.info(`Creating copy of press-release ${collaborationActivity.pressReleaseUri} to temporary graph ${tempGraph}`);
+    await copyPressReleaseToTemporaryGraph(collaborationActivity.pressReleaseUri, tempGraph);
+
+    for (const collaborator of collaborators) {
+      // for every collaborator linked to the collaboration activity, the temporary graph is copied.
+      const target = `${COLLABORATOR_GRAPH_PREFIX}${collaborator.id}`;
+      console.info(`Moving data from temporary graph to collaborator graph ( ${target} )`);
+      await moveGraph(tempGraph, target);
+    }
+
+    // remove temporary graph
+    console.info(`Removing temporary graph ( ${tempGraph} )`);
+    await removeGraph(tempGraph);
+
+    res.sendStatus(204);
+  } catch (err) {
+    console.error(err);
+    next(err);
+  }
 });
 
+app.post('/collaboration-activities/:id/claims', async (req, res, next) => {
+  try {
+    const collaborationActivityId = req.params.id;
+    const collaborationActivity = await getCollaborationActivityById(collaborationActivityId);
+    if (!collaborationActivity) {
+      // 404 if collaboration-activity with provided id does not exist
+      return res.sendStatus(404);
+    }
+
+    if (collaborationActivity.tokenClaimUri) {
+      // 409 if collaboration-activity already has a token-claim linked to it.
+      return res.sendStatus(409);
+    }
+
+    // check if request is made by a user that s part of an organization that participates in the collaboration-activity
+    const requestedByOrganization = await getOrganizationFromHeaders(req.headers);
+    const collaborators = await getCollaborators(collaborationActivity.uri);
+    if (collaborators.map(collaborator => collaborator.uri).indexOf(requestedByOrganization.uri) === -1) {
+      return res.sendStatus(403);
+    }
+
+    // Get user URI from headers to link to token-claim (prov:wasAttributedTo)
+    const claimingUser = await getUserFromHeaders(req.headers);
+
+    // Generate new uri and created date for tokenClaim
+    const newTokenClaimURI = 'http://themis.vlaanderen.be/id/tokenclaim/' + generateUuid();
+    const createdDate = new Date();
+
+    for (const collaborator of collaborators) {
+      // for every collaborator linked to the collaboration activity, the token claim is added to their graphs.
+      const targetGraph = `${COLLABORATOR_GRAPH_PREFIX}${collaborator.id}`;
+      console.info(`Creating token-claim data in collaborator graph ( ${targetGraph} )`);
+      try {
+        await createTokenClaim(
+          newTokenClaimURI,
+          createdDate,
+          claimingUser.uri,
+          collaborationActivity.uri,
+          targetGraph,
+        );
+        console.info(`Successfully created token-claim data in collaborator graph ( ${targetGraph} )`);
+      } catch (err) {
+        console.error(`Failed to create token-claim data in collaborator graph ( ${targetGraph} )`);
+        return handleGenericError(err, next);
+      }
+    }
```

Comment:
Hier zou ik niet loopen over alle collaborators, maar de SPARQL query op zo'n manier herschrijven dat de token-claim geinsert wordt in alle graphs die ook de collaboration-activity bevat (dus een `INSERT ... WHERE` met een variable graph ipv `INSERT DATA` met een fixed graph. Op die manier is er maar 1 SPARQL query nodig om de token-claim beschikbaar te stellen voor alle collaborators.

Hierdoor gaan ook sommige zaken zoals het genereren van de URI en creation-date kunnen verhuizen naar de `createTokenClaim()` method waardoor de `app.js` weer iets korter/overzichtelijker wordt.